### PR TITLE
Encoding shared slices of arrays.

### PR DESCRIPTION
**Pull request by eddyxu**

### CODE REVIEW

File: rust/src/io/writer.rs [hunk 1]
```diff
@@ -198,6 +198,11 @@ impl<'a> FileWriter<'a> {
     async fn write_binary_array(&mut self, field: &Field, array: &ArrayRef) -> Result<()> {
         assert_eq!(field.encoding, Some(Encoding::VarBinary));
         let mut encoder = BinaryEncoder::new(&mut self.object_writer);
+        println!(
```
Comment (changhiskhan) [inline rust/src/io/writer.rs:201]: debug print?

File: rust/src/encodings/plain.rs [hunk 2]
```diff
@@ -81,8 +80,15 @@ impl<'a> PlainEncoder<'a> {
     /// Encode primitive values.
     async fn encode_primitive(&mut self, array: &dyn Array) -> Result<usize> {
         let offset = self.writer.tell();
-        let data = array.data().buffers()[0].as_slice();
-        self.writer.write_all(data).await?;
+        let data = array.data();
+        if matches!(array.data_type(), DataType::Boolean) {
+            self.writer.write_all(&data.buffers()[0].as_slice()).await?;
```
Comment (changhiskhan) [inline rust/src/encodings/plain.rs:85]: is there test coverage for this already?

Comment (eddyxu) [reply rust/src/encodings/plain.rs]: added a test.

File: rust/src/io/writer.rs (hunk 1 above)
Comment (eddyxu) [reply rust/src/io/writer.rs]: Removed

**Review by changhiskhan — APPROVED** [review]

**Pull request merged**